runtime/list.go

```diff
@@ -306,6 +306,70 @@ func listNE(f *Frame, v, w *Object) (*Object, *BaseException) {
 	return listCompare(f, toListUnsafe(v), w, NE)
 }
 
+func listIndex(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+	expectedTypes := []*Type{ListType, ObjectType, IntType, IntType}
+	l := toListUnsafe(args[0])
+	var start, stop int
+	l.mutex.RLock()
+	switch len(args) {
+	case 2:
+		expectedTypes = expectedTypes[:2]
+		start = 0
+		stop = len(l.elems)
+	case 3:
+		expectedTypes = expectedTypes[:3]
+		numElems := len(l.elems)
+		start = toIntUnsafe(args[2]).Value()
+		if start < 0 {
+			start += numElems
+		}
+		stop = numElems
+	case 4:
+		expectedTypes = expectedTypes[:4]
+		numElems := len(l.elems)
+		start = toIntUnsafe(args[2]).Value()
+		if start < 0 {
+			start += numElems
+		}
+		stop = toIntUnsafe(args[3]).Value()
+		if stop < 0 {
+			stop += numElems
+		}
+		if stop > numElems {
+			stop = numElems
+		}
+	}
+	if raised := checkMethodArgs(f, "index", args, expectedTypes...); raised != nil {
+		l.mutex.RUnlock()
+		return nil, raised
+	}
+	value := args[1]
+	index := 0
+	found := false
+	var raised *BaseException
+	for i := start; i < stop; i++ {
```

Comment:
This seems very similar to the implementation of remove() so maybe it could be factored out into a helper function in seq.go that takes a slice and returns the index.